src/components/NurseryWithdrawals/PlantingProgressMap.tsx

```diff
@@ -1,19 +1,52 @@
-import { useAppSelector } from 'src/redux/store';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
 import { selectPlantingSite } from 'src/redux/features/tracking/trackingSelectors';
 import { MapService } from 'src/services';
-import { MapData } from 'src/types/Map';
-import { useEffect, useState } from 'react';
+import { MapData, MapSourceProperties } from 'src/types/Map';
+import { useEffect, useMemo, useState } from 'react';
 import { PlantingSiteMap } from 'src/components/Map';
 import { Typography, useTheme } from '@mui/material';
+import PlantingProgressMapDialog from 'src/components/NurseryWithdrawals/PlantingProgressMapDialog';
+import { makeStyles } from '@mui/styles';
+import { requestSitePopulation } from 'src/redux/features/tracking/trackingThunks';
+import { useOrganization } from 'src/providers';
+
+export const useStyles = makeStyles(() => ({
+  popup: {
+    '& > .mapboxgl-popup-content': {
+      borderRadius: '8px',
+      padding: '0',
+      minWidth: '320px',
+    },
+  },
+}));
 
 type PlantingProgressMapProps = {
   plantingSiteId: number;
 };
 
 export default function PlantingProgressMap({ plantingSiteId }: PlantingProgressMapProps): JSX.Element {
   const theme = useTheme();
+  const classes = useStyles();
+  const dispatch = useAppDispatch();
+  const org = useOrganization();
   const plantingSite = useAppSelector((state) => selectPlantingSite(state, plantingSiteId));
   const [mapData, setMapData] = useState<MapData | undefined>();
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [lastPlantingSiteId, setLastPlantingSiteId] = useState<number | undefined>();
+  const [focusEntities, setFocusEntities] = useState<{ sourceId: string; id: number }[]>();
+  useEffect(() => {
+    // set focus entities to the planting site if the planting site id has changed
+    setLastPlantingSiteId((last) => {
+      if (plantingSiteId !== last) {
+        setFocusEntities([{ sourceId: 'sites', id: plantingSiteId }]);
+      } else {
+        setFocusEntities([]);
+      }
+      return plantingSiteId;
+    });
+  }, [plantingSiteId, plantingSite]);
```

Comment:
do we need the plantingSite? Listening to object changes in useEffects have a lot of unwanted triggers. We should listen to specific properties  within the object as needed.

Reply:
I meant 'plantingSite' it is not used anywhere in the code. If it is just to trigger the useEffect maybe we could just listen for plantingSite.id ?

Reply:
>I meant 'plantingSite' it is not used anywhere in the code. If it is just to trigger the useEffect maybe we could just listen for plantingSite.id ?

Maybe! Lemme think about this a little

Reply:
That's a decent idea that might avoid a lot of this complexity. I'll give that a try. Thanks

Reply:
Oh wait.. there’s no guarantee the refresh planting sites would have completed by then.